documentation/modules/exploit/windows/local/tokenmagic.md

```diff
@@ -0,0 +1,267 @@
+## Vulnerable Application
+This module exploits a UAC bypass in windows that allows the attacker to obtain 
+remote code execution by leveraged a privileged file write. From the [PoC](https://github.com/FuzzySecurity/PowerShell-Suite/blob/master/UAC-TokenMagic.ps1):
+
+>Essentially we duplicate the token of an elevated process, lower it's mandatory
+integrity level, use it to create a new restricted token, impersonate it and
+use the Secondary Logon service to spawn a new process with High IL. Like
+playing hide-and-go-seek with tokens
+
+The module exploits the high IL gained from the "token magic" by either starting a malicious service or by preforming a DLL hijack 
+on a known DLL in system32.
+
+### Installation And Setup
+Windows 10 versions 1803 is vulnerable out of the box. Token Magic works on Windows 7, 8, 8,1 and Windows 10 instances up to 1803. The DLL 
+hijacking method in this module relies on a DLL that is only usable in Windows 1703 - 1803 and will not work on other 
+versions. Also note the DLL method uses a trigger that can take up to ten minutes to return a shell. 
+
+## Verification Steps
+ 1. Start msfconsole
+ 2. Get a Meterpreter session
+
+Run the token magic exploit:
+3. Do: `use exploit/windows/local/tokenmagic`
+4. Set the `METHOD` of exploitation, either `DLL` or `SERVICE`
+5. Set the `LHOST`, `SESSION` and `PAYLOAD` options
+6. Do: `run`
+7. You should get a shell, the exploitation process should be fairly instantaneous 
+
+## Options
+  **METHOD**
+  Select between DLL hijacking and service exploitation
+  DLL mode: Using the elevated privileges from token magic the module will write a malicious file to c:\windows\system32\windowscoredeviceinfo.dll, a notepad.exe process is spawned and a DLL trigger is injected into the process to call the `usoclient`. When the `usoclient` runs it loads the the malicious DLL windowscoredeviceinfo.dll with SYSTEM level privileges.
+  SERVICE mode: Using the elevated privileges from token magic the module create a malicious service and then start it with SYSTEM level privileges
+
+  **SERVICE_FILENAME**  
+  Filename for Service Payload (Random by default).
+
+  **SERVICE_NAME**      
+  Service Name to use (Random by default).
+
+  **SESSION**           
+  The session to run this module on.
+
+  **WRITABLE_DIR** 
+  Directory to write file to (%TEMP% by default).
```

Comment:
```suggestion
  Directory to write file to (`%TEMP%` by default).
```

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/15168/commits/8c3e35822e093b91503f98ddf66b4e6218fba083